fud/fud/stages/verilator/numeric_types.py

```diff
@@ -97,21 +101,22 @@ class Bitnum(NumericType):
     def __init__(self, value: str, width: int, is_signed: bool):
         super().__init__(value, width, is_signed)
 
-        if self.bit_string_repr is None and self.hex_string_repr is None:
-            # The decimal representation was passed in.
-            self.bit_string_repr = np.binary_repr(int(self.string_repr), self.width)
-            self.uint_repr = int(self.bit_string_repr, 2)
-            self.hex_string_repr = np.base_repr(self.uint_repr, 16)
-
-        if is_signed and self.uint_repr > (2 ** (width - 1)):
-            negated_value = -1 * ((2 ** width) - self.uint_repr)
-            self.string_repr = str(negated_value)
-
-        if len(self.bit_string_repr) > width:
-            raise InvalidNumericType(
-                f"The value: {value} will overflow when trying to represent "
-                f"{len(self.bit_string_repr)} bits with width: {width}"
-            )
+        if not self.is_undef:
```

Comment:
Nit: can we just do an early return here for the `undef` case, instead of nesting the rest of the code another level deeper?

Reply:
That’s a good idea! Unfortunately looks like auto merge already merged this in. Feel free to open a PR to make the change